File: dia_001/two_sum.js
function two_sum(nums, target) {
  const resultado = {};
  for (let i = 0; i < nums.length; i++) {
    const complemento = target - nums[i];
    if (complemento in resultado) {
      return [resultado[complemento], i];
    }
    resultado[nums[i]] = i;
  }
}

const nums = [3, 2, 4];
const target = 6;

const resultado = two_sum(nums, target);
console.log(resultado);

File: dia_001/two_sum.java
// Versão da IDE, VSCode e IntelliJ
package dia_001;

import java.util.*;

class Main {
    public static List<Integer> two_sum(List<Integer> nums, int target) {
        Map<Integer, Integer> resultado = new HashMap<>();
        for (int i = 0; i < nums.size(); i++) {
            int complemento = target - nums.get(i);
            if (resultado.containsKey(complemento)) {
                return Arrays.asList(resultado.get(complemento), i);
            }
            resultado.put(nums.get(i), i);
        }
        return new ArrayList<>();
    }


    public static void main(String[] args) {
        int target = 9;
        List<Integer> nums = Arrays.asList(2,7,11,15);
        List<Integer> resultado = Main.two_sum(nums, target);
        System.out.println(resultado);
    }
}

// Versão para o LeetCode: 
/*
class Solution {
    public int[] twoSum(int[] nums, int target) {
Map<Integer, Integer> resultado = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complemento = target - nums[i];
            if (resultado.containsKey(complemento)) {
                return new int[] { resultado.get(complemento), i };
            }
            resultado.put(nums[i], i);
        }
        return new int[] {};
}}
*/